test/models/rhsm_fact_parser_test.rb

```diff
@@ -162,6 +162,17 @@ def test_operatingsystem_almalinux
       assert_equal parser.operatingsystem.minor, '3'
     end
 
+    def test_operatingsystem_amazon
+      @facts['distribution.name'] = 'Amazon'
+      @facts['distribution.version'] = '2.2'
+      @facts['distribution.id'] = 'Karoo'
+
+      assert_equal parser.operatingsystem.name, 'Amazon'
+      assert_equal parser.operatingsystem.type, 'Redhat'
+      assert_equal parser.operatingsystem.major, '2'
+      assert_equal parser.operatingsystem.minor, '2'
```

Comment:
This is failing: 

Expected: nil
  Actual: "Redhat" (Minitest::Assertion)
/home/jenkins/workspace/katello-pr-test/test/models/rhsm_fact_parser_test.rb:171